pkg/apis/nfd/nodefeaturerule/expression.go

```diff
@@ -337,23 +337,152 @@ func MatchGetValues(m *nfdv1alpha1.MatchExpressionSet, values map[string]string)
 // features, each of which is an individual set of key-value pairs
 // (attributes).
 func MatchInstances(m *nfdv1alpha1.MatchExpressionSet, instances []nfdv1alpha1.InstanceFeature) (bool, error) {
-	v, err := MatchGetInstances(m, instances)
-	return len(v) > 0, err
+	isMatch, _, err := MatchGetInstances(m, instances)
+	return isMatch, err
 }
 
 // MatchGetInstances evaluates the MatchExpressionSet against a set of instance
 // features, each of which is an individual set of key-value pairs
 // (attributes). A slice containing all matching instances is returned. An
 // empty (non-nil) slice is returned if no matching instances were found.
```

Comment:
doc needs to be updated

Reply:
You refer to the return values? Fixed that part.